msbuild/Xamarin.iOS.Tasks.Core/Tasks/MTouchTaskBase.cs

```diff
@@ -573,14 +576,31 @@ protected override string GenerateCommandLineCommands ()
 				args.AddQuoted (gcc.Arguments.ToString ());
 			}
 
-			foreach (var asm in References) {
-				if (IsFrameworkItem(asm)) {
-					args.AddQuoted ("-r=" + ResolveFrameworkFile(asm.ItemSpec));
-				} else {
-					args.AddQuoted ("-r=" + Path.GetFullPath (asm.ItemSpec));
+			// Generate a response file
+			var responseFile = Path.GetFullPath (ResponseFilePath);
+
+			if (File.Exists (responseFile))
+				File.Delete (responseFile);
+
+			try {
+				using (var fs = File.Create (responseFile)) {
+					using (var writer = new StreamWriter (fs)) {
+						foreach (var asm in References) {
+							if (IsFrameworkItem (asm)) {
+								writer.Write (" -r=" + ResolveFrameworkFile (asm.ItemSpec));
```

Comment:
IIRC it's one argument per line, and it looks like you're putting them all in a single line.

Reply:
It works like a charm though...

Reply:
Did you try paths with spaces? :smile:

Reply:
@rolfbjarne spaced in the references?

Reply:
Yes, a reference with spaces in its path.

Reply:
Ok indeed it breaks. Oops 😅

Reply:
Well it looks like it's more the lack of quotes than the fact it's all in a single line

Reply:
It'll be much easier to read the response file (for humans) if each argument is on a new line 😄